polkadot/xcm/xcm-builder/src/routing.rs

```diff
@@ -149,12 +153,21 @@ impl EnsureDelivery for Tuple {
 /// Inspects messages in queues.
 /// Meant to be used in runtime APIs, not in runtimes.
 pub trait InspectMessageQueues {
+	/// Clear the queues at the beginning of Runtime API call, so that subsequent
+	/// `Self::get_messages()` will return only messages generated by said Runtime API.
+	fn clear_messages() {}
```

Comment:
Had to add a default implementation so it's not a breaking change since this is a public trait.

Reply:
So we can backport it to stable2409

Reply:
You could add the default impl specifically only in the Backport MR. Then it would not have a default on master.

Reply:
Sounds good